lib/elixir/lib/calendar/time.ex

```diff
@@ -358,6 +359,59 @@ defmodule Time do
     end
   end
 
+  @doc """
+  Converts a number of seconds after midnight to a `Time` struct.
+
+  ## Examples
+
+      iex> Time.from_seconds_after_midnight(10_000)
+      ~T[02:46:40]
+      iex> Time.from_seconds_after_midnight(30_000, {5000, 3})
+      ~T[08:20:00.005]
+      iex> Time.from_seconds_after_midnight(-1)
+      ~T[23:59:59]
+      iex> Time.from_seconds_after_midnight(100_000)
+      ~T[03:46:40]
+
+  """
+  @spec from_seconds_after_midnight(
+          integer(),
+          Calendar.microsecond(),
+          Calendar.calendar()
+        ) :: t
+  def from_seconds_after_midnight(seconds, microsecond \\ {0, 0}, calendar \\ Calendar.ISO)
+      when is_integer(seconds) do
+    seconds_in_day = Integer.mod(seconds, @seconds_per_day)
+
+    {hour, minute, second, {_, _}} =
+      calendar.time_from_day_fraction({seconds_in_day, @seconds_per_day})
+
+    %Time{
+      calendar: calendar,
+      hour: hour,
+      minute: minute,
+      second: second,
+      microsecond: microsecond
+    }
+  end
+
+  @doc """
+  Converts a `Time` struct to a number of seconds after midnight.
+
+  ## Examples
+
+      iex> Time.to_seconds_after_midnight(~T[23:30:15])
+      84615
+      iex> Time.to_seconds_after_midnight(~N[2010-04-17 23:30:15.999])
+      84615
+
+  """
+  @spec to_seconds_after_midnight(t) :: Calendar.seconds_after_midnight()
```

Comment:
similar to previous comment, `t()` vs `Calendar.time()`.

Also, there's no `Calendar.seconds_after_midnight()` type, should we use `integer()` since that's what we use in `from_seconds_after_midnight/3` too?

```suggestion
  @spec to_seconds_after_midnight(Calendar.time()) :: integer()
```

Reply:
Right, thanks!